master/internal/rm/kubernetesrm/pods.go

```diff
@@ -211,85 +192,254 @@ func Initialize(
+		containerID: msg.Spec.ContainerID,
+	}
+
+	return nil
+}
+
+func (p *pods) ChangePriority(podID cproto.ID) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	ref := p.verifyPodAndGetRef(string(podID))
+	if ref == nil {
+		return
+	}
+	ref.ChangePriority()
+}
+
+func (p *pods) ChangePosition(podID cproto.ID) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	ref := p.verifyPodAndGetRef(string(podID))
+	if ref == nil {
+		return
+	}
+	ref.ChangePosition()
+}
+
+func (p *pods) KillTaskPod(podID cproto.ID) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	podIDStr := string(podID)
+	ref := p.verifyPodAndGetRef(podIDStr)
+	if ref == nil {
+		return
+	}
+	ref.KillTaskPod()
+}
+
+// ResourceSummaryResponse contains information for pods.
+type ResourceSummaryResponse struct {
+	NumAgents      int
+	SlotsAvailable int
+}
+
+func (p *pods) SummarizeResources(poolName string) (*ResourceSummaryResponse, error) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	summary, err := p.summarize()
+	if err != nil {
+		return nil, err
+	}
+
+	slots := 0
+	if len(poolName) > 0 {
+		slots = numSlots(summary[poolName].Slots)
+	} else {
+		for _, pool := range summary {
+			slots += numSlots(pool.Slots)
 		}
+	}
+	return &ResourceSummaryResponse{NumAgents: len(summary), SlotsAvailable: slots}, nil
+}
 
-	case actor.PostStop:
+type reattachPodsRequest struct {
+	numPods      int
+	allocationID model.AllocationID
+	slots        int
+	logContext   logger.Context
+}
 
-	case StartTaskPod:
-		if err := p.receiveStartTaskPod(ctx, msg); err != nil {
-			return err
+type reattachPodResponse struct {
+	containerID string
+	started     *sproto.ResourcesStarted
+}
+
+func (p *pods) reattachPods(req reattachPodsRequest) ([]reattachPodResponse, error) {
+	p.mu.Lock()
+	defer p.mu.Unlock()
+
+	listOptions := metaV1.ListOptions{
+		LabelSelector: fmt.Sprintf("%s=%s", determinedLabel, req.allocationID),
+	}
+
+	pods, err := p.listPodsInAllNamespaces(context.TODO(), listOptions)
+	if err != nil {
+		return nil, fmt.Errorf("error listing pods checking if they can be restored: %w", err)
+	}
+
+	configMaps, err := p.listConfigMapsInAllNamespaces(context.TODO(), listOptions)
+	if err != nil {
+		return nil, fmt.Errorf("error listing config maps checking if they can be restored: %w", err)
+	}
+	existingConfigMaps := make(set.Set[string])
+	for _, cm := range configMaps.Items {
+		if _, ok := p.namespaceToPoolName[cm.Namespace]; !ok {
+			continue
 		}
+		existingConfigMaps.Insert(cm.Name)
+	}
 
-	case ChangePriority:
-		p.receivePriorityChange(ctx, msg)
+	var containerIDs []string
+	var k8sPods []*k8sV1.Pod
+	var ports [][]int
+	var resourcePool string
+	for _, pod := range pods.Items {
+		if _, ok := p.namespaceToPoolName[pod.Namespace]; !ok {
+			continue
+		}
 
-	case ChangePosition:
-		p.receivePositionChange(ctx, msg)
+		foundID := false
+		foundPool := false
+		for _, container := range pod.Spec.Containers {
+			for _, env := range container.Env {
+				switch env.Name {
+				case "DET_CONTAINER_ID":
```

Comment:
nit: I wonder if using pod labels & selectors for filtering would be more idiomatic